source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -3305,3 +3305,826 @@ BOOST_AUTO_TEST_CASE(impact_with_timeslots_compare_timeslots_diff_size) {
+                "stopE", "08:40"_t)("stopF", "08:50"_t);
+        b.vj("line:1", "111111", "", true, "vj:A-2")
+            .route("route-2")("stopF", "08:00"_t)("stopE", "08:10"_t)("stopD", "08:20"_t)("stopC", "08:30"_t)(
+                "stopB", "08:40"_t)("stopA", "08:50"_t);
+    });
+
+    b.data->meta->production_date = bg::date_period(bg::date(2021, 1, 1), bg::days(7));
+
+    using btp = boost::posix_time::time_period;
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 2);
+
+    // new rail_section disruption
+    navitia::apply_disruption(
+        b.impact(nt::RTLevel::Adapted, "rail_section_on_line1")
+            .severity(nt::disruption::Effect::NO_SERVICE)
+            .application_periods(btp("20210101T060000"_dt, "20210101T100000"_dt))
+            .on_rail_section("line:1", "stopAreaC", "stopAreaF",
+                             {std::make_pair("stopAreaE", 3), std::make_pair("stopAreaD", 2)}, {}, *b.data->pt_data)
+            .get_disruption(),
+        *b.data->pt_data, *b.data->meta);
+
+    // Check every relevant fields
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->routes.size(), 2);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 3);
+
+    // Check the original vj
+    auto* vj = b.get<nt::VehicleJourney>("vehicle_journey:vj:A-1");
+    auto adapted_vp = vj->adapted_validity_pattern()->days;
+    auto base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "111111"), base_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "111110"), adapted_vp);
+    vj = b.get<nt::VehicleJourney>("vehicle_journey:vj:A-2");
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "111111"), base_vp);
+
+    // Check the adapted vj
+    vj = b.get<nt::VehicleJourney>("vehicle_journey:vj:A-1:Adapted:0:rail_section_on_line1");
+    adapted_vp = vj->adapted_validity_pattern()->days;
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "000001"), adapted_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "000000"), base_vp);
+    // The adapted vj should have only 2 stop_times, for A, B and C stop point
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.size(), 3);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list[0].stop_point->uri, "stopA");
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list[1].stop_point->uri, "stopB");
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list[2].stop_point->uri, "stopC");
+
+    // disruption
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->disruption_holder.nb_disruptions(), 1);
+    BOOST_REQUIRE_EQUAL(b.get<nt::StopPoint>("stopA")->get_impacts().size(), 0);
+    BOOST_REQUIRE_EQUAL(b.get<nt::StopPoint>("stopB")->get_impacts().size(), 0);
+    BOOST_REQUIRE_EQUAL(b.get<nt::StopPoint>("stopC")->get_impacts().size(), 1);
```

Comment:
I'm not sure we should impact this stop "StopC", just `PICK_UP=False`

Reply:
the discussion is in progress

Reply:
We said, we impact all stop points and C is inside